### PR TITLE
Feature/device headers files

### PR DESCRIPTION
**Pull request by salkinium**
- Adds up-to-date cmsis header files for all STM32 devices as a git subtree from [modm-io/cmsis-header-stm32](https://github.com/modm-io/cmsis-header-stm32).
- Adds all available device files to avoid future manual device file porting (relates to #221).
- Adds documentation to deprecate the DFG in favor of the one in [modm-devices](https://github.com/modm-io/modm-devices).

I didn't want to add a git submodule, since there is no way to clone recursively by default.
cc @dergraaf @strongly-typed @ekiwi how do you feel about using git subtree here?

### CODE REVIEW

File: ext/st/README.md [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
+- [CubeF4 v1.14.0 (v2.5.0) released 22-April-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubef4.html)
+- [CubeF7 v1.6.0 (v1.1.0) released 22-April-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubef7.html)
+- [CubeL0 v1.8.0 (v1.7.1) released 25-November-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubel0.html)
+- [CubeL1 v1.6.0 (v2.2.0) released 01-July-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubel1.html)
+- [CubeL4 v1.6.0 (v1.8.0) released 28-October-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubel4.html)
```
Comment (strongly-typed) [inline ext/st/README.md]: STM32Cube_FW_L4_V1.7.0 is available with V1.3.0 released 17-February-2017.

Comment (salkinium) [reply ext/st/README.md]: Hm, and also (v1.8) is completely wrong. Was probably too worked up by all the trailing whitespace to do this properly.

File: ext/st/README.md [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
+Here is the list of the current Cube version, device header version in braces and release date of the device headers:
+
+- [CubeF0 v1.7.0 (v2.3.1) released 04-November-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubef0.html)
+- [CubeF1 v1.4.0 (v4.1.0) released 29-April-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubef1.html)
+- [CubeF2 v1.4.0 (v2.1.2) released 29-June-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubef2.html)
+- [CubeF3 v1.7.0 (v2.3.1) released 16-December-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubef3.html)
+- [CubeF4 v1.14.0 (v2.5.0) released 22-April-2016](http://www.st.com/content/st_com/en/products/embedded-software/mcus-embedded-software/stm32-embedded-software/stm32cube-embedded-software/stm32cubef4.html)
```
Comment (strongly-typed) [inline ext/st/README.md]: STM32Cube_FW_F4_V1.15.0 is available with V2.6.1 released 14-February-2017.

**Comment by strongly-typed** [comment]
I did not use git subtree extensively. But moving all external dependencies in an external repos is highly appreciated here (including fatfs, FreeRTOS, ...). So +1 git subtree

File: ext/st/README.md (hunk 2 above)
Comment (salkinium) [reply ext/st/README.md]: All headers are updated to the latest version, will reimport in a second.
[cmsis-header-stm32](https://github.com/modm-io/cmsis-header-stm32) now checks daily for new versions thanks to Travis CI! [Check it out!](https://travis-ci.org/modm-io/cmsis-header-stm32/builds/212266527)

**Comment by salkinium** [comment]
@strongly-typed Added the L4 and all other device files.

**Comment by salkinium** [comment]
This PR broke a lot of things, since I updated the device headers as well as the device files. It will require a little bit of work.
Furthermore, even though the {F2, L0, L1, L4} families have been added, the build system is unaware of these devices for now. This support will be added when needed.

**Comment by salkinium** [comment]
Any objections for merging this?

**Comment by strongly-typed** [comment]
Did you test this with any hardware? I won't have access to any non-L4 hardware before Monday evening.

**Comment by salkinium** [comment]
> Did you test this with any hardware?

Yes, all my Nucleo and Discovery boards do what they are supposed to do.

The changes to existing device files are minimal, but there are some changes to the device headers from ST, that may break things for existing applications. But that's not really within my control, tbh.

**Comment by strongly-typed** [comment]
I'm fine with merging. Everything compiles for me.

**Pull request merged**